libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesJobIntentService.java

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.android.telemetry.location;
+
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.support.annotation.NonNull;
+import android.support.v4.app.JobIntentService;
+import android.util.Log;
+import com.mapbox.android.core.location.LocationEngineResult;
+import com.mapbox.android.telemetry.LocationEvent;
+
+import java.util.List;
+
+/**
+ * This job intent service is responsible for managing location updates
+ * in the background since the updates are reported via broadcast mechanism
+ * and therefore are dispatched on the main thread.
+ */
+public class LocationUpdatesJobIntentService extends JobIntentService {
+  private static final int JOB_ID = 102;
+  private static final String TAG = "LocJobIntentService";
+
+  public static void enqueueWork(Context context, Intent intent) {
+    enqueueWork(context, LocationUpdatesJobIntentService.class, JOB_ID, intent);
+  }
+
+  @Override
+  protected void onHandleWork(@NonNull Intent intent) {
```

Comment:
It seems that we don't have time consuming work here, so why not do these works in `LocationUpdatesBroadcastReceiver` directly?

Reply:
I agree it might be an overkill, will revisit in the next iteration, i did not write any unit tests for it for the reason that i'd be profiling it right after i integrate it with mapbox telem.